src/server.jl

```diff
@@ -199,6 +202,13 @@ function _extract_relevant_options(file_frontmatter::Dict, options::Dict)
         julia = get(metadata, "julia", Dict())
         julia_merged = _recursive_merge(julia_default, julia)
 
+        # quarto stores params in two places currently, in `format.metadata.params` we have params specified in the front matter
+        # and in top-level `params` we have the parameters via command-line arguments.
+        # In case quarto decides to unify this behavior later, we probably can stop merging these on our side.
+        # Cf. https://github.com/quarto-dev/quarto-cli/issues/9197
+        params = get(metadata, "params", Dict())
+        cli_params = get(options, "params", Dict())
+        params_merged = _recursive_merge(params_default, params, cli_params)
```

Comment:
Should we be validating that the keys are all valid Julia identifiers with `Base.isidentifier`?

Reply:
Good idea